builder/rootfs/bin/boot

```diff
@@ -14,14 +14,16 @@ fi
 [[ $DEBUG ]] && set -x
 
 # configure etcd
-export ETCD_PORT=${ETCD_PORT:-4001}
-export ETCD="$HOST:$ETCD_PORT"
+# if ETCDCTL_PEERS isn't already set, set it
+if [ -z "$ETCDCTL_PEERS" ]; then
+	export ETCDCTL_PEERS="http://${ETCD_HOST:-${HOST:-127.0.0.1}}:${ETCD_PORT:-4001}"
```

Comment:
For backwards compatibility, I was trying to always fall back to how things have been done thus far if the `ETCDCTL_PEERS` env variable isn't set, and that meant relying on all the other env variables you see here.  It's not ideal and even a bit confusing to understand.  I do think that on our next major release we should have a big simplification / refactor of this where the URL as a whole can be configured as @bacongobbler suggests.